index.html

```diff
@@ -244,14 +217,30 @@
 
-      <p>The decision to cancel a <code><a>preconnect</a></code> or <code><a>speculative preload</a></code> fetch is deferred to the user agent, which MAY use local context and other signals to determine if and when the optimization should be aborted:<p>
+      <p>The decision to cancel a <code><a>preconnect</a></code> or <code><a>speculative fetch</a></code> is deferred to the user agent, which MAY use local context and other signals to determine if and when the optimization should be aborted:<p>
 
       <ul>
         <li>The user agent SHOULD abort the request if the `href` attribute on the `link` element of an <a>resource hint link</a> is removed, or its value is set to an empty string.</li>
-        <li>The user agent SHOULD allow the requests initiated via the <a>resource hint link</a> to continue across navigations. However, the user agent MAY cancel these requests once it determines that they are not required by the new destination - see <a href="#reactive-resource-prefetching-preload"></a>.</p></li>
+        <li>The user agent SHOULD allow the requests initiated via the <a>resource hint link</a> to continue across navigations. However, the user agent MAY cancel these requests once it determines that they are not required by the new destination - see <a href="#reactive-resource-prefetching-prefetch"></a>.</p></li>
       </ul>
+
+      <p>Resource fetches that may be required for the next navigation can negatively impact the performance of the current navigation context due to additional contention for the CPU, GPU, memory, and network resources. To address this, the user agent SHOULD implement logic to reduce and eliminate such contention:</p>
+
+      <ul>
+        <li>Resource fetches required for the next navigation SHOULD have lower relative priority and SHOULD NOT block or interfere with resource fetches required by the current navigation context.</li>
+
+        <li>The optimal time to initiate a resource fetch required for the next navigation is dependent on the negotiated transport protocol, users current connectivity profile, available device resources, and other context specific variables. The user agent is left to determine the optimal time at which to initiate the fetch - e.g. the user agent MAY decide to wait until all other downloads are finished, or MAY choose to pipeline requests with low priority if the negotiated protocol supports the necessary primitives. Alternatively, the user agent MAY opt-out from initiating the fetch due to resource constraints, user preferences, or other factors.</li>
+      </ul>
+    </section>
+
+    <section>
+      <h2>Load and error events</h2>
+
+      <p>The user agent MUST NOT <a href="https://html.spec.whatwg.org/#delay-the-load-event">delay the `load` event</a> of the document due to outstanding <code><a>preconnect</a></code> or <code><a>speculative fetch</a></code> initiated requests.</p>
```

Comment:
What happens with outstanding `preload` fetches? We mustn't block on them either IMO (unless the parser encountered the elements that require them before onload)